controllers/awsmachinetemplate_controller.go

```diff
@@ -0,0 +1,259 @@
+			}
+		} else if awsCluster.Spec.Region != "" {
+			return awsCluster.Spec.Region, nil
+		}
+	}
+
+	return "", nil
+}
+
+// getInstanceTypeInfo queries AWS EC2 API for instance type capacity and node info.
+func (r *AWSMachineTemplateReconciler) getInstanceTypeInfo(ctx context.Context, globalScope *scope.GlobalScope, template *infrav1.AWSMachineTemplate, instanceType string) (corev1.ResourceList, *infrav1.NodeInfo, error) {
+	// Create EC2 client from global scope
+	ec2Client := ec2.NewFromConfig(globalScope.Session())
+
+	// Query instance type information
+	input := &ec2.DescribeInstanceTypesInput{
+		InstanceTypes: []ec2types.InstanceType{ec2types.InstanceType(instanceType)},
+	}
+
+	result, err := ec2Client.DescribeInstanceTypes(ctx, input)
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "failed to describe instance type %q", instanceType)
+	}
+
+	if len(result.InstanceTypes) == 0 {
+		return nil, nil, errors.Errorf("no information found for instance type %q", instanceType)
+	}
+
+	// Extract capacity information
+	info := result.InstanceTypes[0]
+	resourceList := corev1.ResourceList{}
+
+	// CPU
+	if info.VCpuInfo != nil && info.VCpuInfo.DefaultVCpus != nil {
+		resourceList[corev1.ResourceCPU] = *resource.NewQuantity(int64(*info.VCpuInfo.DefaultVCpus), resource.DecimalSI)
+	}
+
+	// Memory
+	if info.MemoryInfo != nil && info.MemoryInfo.SizeInMiB != nil {
+		memoryBytes := *info.MemoryInfo.SizeInMiB * 1024 * 1024
+		resourceList[corev1.ResourceMemory] = *resource.NewQuantity(memoryBytes, resource.BinarySI)
+	}
+
+	// Extract node info from AMI if available
+	nodeInfo := &infrav1.NodeInfo{}
+	amiID := template.Spec.Template.Spec.AMI.ID
+	if amiID != nil && *amiID != "" {
+		arch, os, err := r.getNodeInfoFromAMI(ctx, ec2Client, *amiID)
+		if err == nil {
+			if arch != "" {
+				nodeInfo.Architecture = arch
+			}
+			if os != "" {
+				nodeInfo.OperatingSystem = os
+			}
+		}
+	}
+
+	return resourceList, nodeInfo, nil
+}
+
+// getNodeInfoFromAMI queries the AMI to determine architecture and operating system.
+func (r *AWSMachineTemplateReconciler) getNodeInfoFromAMI(ctx context.Context, ec2Client *ec2.Client, amiID string) (infrav1.Architecture, string, error) {
+	input := &ec2.DescribeImagesInput{
+		ImageIds: []string{amiID},
+	}
+
+	result, err := ec2Client.DescribeImages(ctx, input)
+	if err != nil {
+		return "", "", errors.Wrapf(err, "failed to describe AMI %q", amiID)
+	}
+
+	if len(result.Images) == 0 {
+		return "", "", errors.Errorf("no information found for AMI %q", amiID)
+	}
+
+	image := result.Images[0]
+
+	// Get architecture from AMI
+	var arch infrav1.Architecture
+	switch image.Architecture {
+	case ec2types.ArchitectureValuesX8664:
+		arch = infrav1.ArchitectureAmd64
+	case ec2types.ArchitectureValuesArm64:
+		arch = infrav1.ArchitectureArm64
+	}
+
+	// Determine OS - check Platform field first (specifically for Windows identification)
+	var os string
```

Comment:
Maybe we could initialize this with `linux` and only change to `windows` if the `image.Platform` or the `image.PlatformDetails` say it's windows? A fairly minor change, but it would eliminate the check at L242.